modules/apps/segments/segments-asah-connector/src/main/java/com/liferay/segments/asah/connector/internal/portlet/action/CalculateSegmentsExperimentEstimatedDurationMVCActionCommand.java

```diff
@@ -0,0 +1,228 @@
+import javax.portlet.ActionResponse;
+
+import javax.servlet.http.HttpServletResponse;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author David Arques
+ */
+@Component(
+	immediate = true,
+	property = {
+		"javax.portlet.name=" + SegmentsPortletKeys.SEGMENTS_EXPERIMENT,
+		"mvc.command.name=/calculate_segments_experiment_estimated_duration"
+	},
+	service = MVCActionCommand.class
+)
+public class CalculateSegmentsExperimentEstimatedDurationMVCActionCommand
+	extends BaseMVCActionCommand {
+
+	@Override
+	protected void doProcessAction(
+			ActionRequest actionRequest, ActionResponse actionResponse)
+		throws Exception {
+
+		ThemeDisplay themeDisplay = (ThemeDisplay)actionRequest.getAttribute(
+			WebKeys.THEME_DISPLAY);
+
+		JSONObject jsonObject = null;
+
+		try {
+			jsonObject = _calculateSegmentsExperimentEstimatedDaysDuration(
+				actionRequest);
+		}
+		catch (Throwable t) {
+			_log.error(t, t);
+
+			HttpServletResponse httpServletResponse =
+				_portal.getHttpServletResponse(actionResponse);
+
+			httpServletResponse.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+
+			jsonObject = JSONUtil.put(
+				"error",
+				LanguageUtil.get(
+					themeDisplay.getRequest(), "an-unexpected-error-occurred"));
+		}
+
+		hideDefaultSuccessMessage(actionRequest);
+
+		JSONPortletResponseUtil.writeJSON(
+			actionRequest, actionResponse, jsonObject);
+	}
+
+	private JSONObject _calculateSegmentsExperimentEstimatedDaysDuration(
+			ActionRequest actionRequest)
+		throws PortalException {
+
+		long segmentsExperimentId = ParamUtil.getLong(
+			actionRequest, "segmentsExperimentId");
+
+		SegmentsExperiment segmentsExperiment =
+			_segmentsExperimentLocalService.getSegmentsExperiment(
+				segmentsExperimentId);
+
+		String segmentsExperimentRels = ParamUtil.getString(
+			actionRequest, "segmentsExperimentRels");
+
+		JSONObject segmentsExperimentRelsJSONObject =
+			JSONFactoryUtil.createJSONObject(segmentsExperimentRels);
+
+		Iterator<String> iterator = segmentsExperimentRelsJSONObject.keys();
+
+		Map<String, Double> segmentsExperienceKeySplitMap = new HashMap<>();
+
+		while (iterator.hasNext()) {
+			String key = iterator.next();
+
+			SegmentsExperimentRel segmentsExperimentRel =
+				_segmentsExperimentRelLocalService.getSegmentsExperimentRel(
+					GetterUtil.getLong(key));
+
+			segmentsExperienceKeySplitMap.put(
+				segmentsExperimentRel.getSegmentsExperienceKey(),
+				segmentsExperimentRelsJSONObject.getDouble(key));
+		}
+
+		Long segmentsExperimentEstimatedDaysDuration =
+			_calculateSegmentsExperimentEstimatedDaysDuration(
+				segmentsExperiment,
+				ParamUtil.getDouble(actionRequest, "confidenceLevel"),
+				segmentsExperienceKeySplitMap);
+
+		return JSONUtil.put(
+			"segmentsExperimentEstimatedDaysDuration",
+			segmentsExperimentEstimatedDaysDuration);
+	}
+
+	private Long _calculateSegmentsExperimentEstimatedDaysDuration(
```

Comment:
Two methods with the same name but returning different things...For me it's confusing

Reply:
`_calculateSegmentsExperimentEstimatedDaysDurationJSONObject`